core-foundation/src/url.rs

```diff
@@ -78,13 +78,15 @@ impl CFURL {
     pub fn to_path(&self) -> Option<PathBuf> {
         // implementing this on Windows is more complicated because of the different OsStr representation
         unsafe {
-            let mut buf: [u8; PATH_MAX as usize] = mem::uninitialized();
-            let result = CFURLGetFileSystemRepresentation(self.0, true as Boolean, buf.as_mut_ptr(), buf.len() as CFIndex);
+            let mut buf = MaybeUninit::<[u8; PATH_MAX as usize]>::uninit();
+            let result = CFURLGetFileSystemRepresentation(self.0, true as Boolean, buf.as_mut_ptr() as *mut u8, PATH_MAX as CFIndex);
```

Comment:
If I don't add `as *mut u8` coercion, then I got an error saying "expected type `*mut u8` but found type `*mut [u8; 1024]`".

Reply:
The type system has saved us here - `MaybeUninit<[u8; N]>::as_mut_ptr()` does not return an equivalent pointer as `[u8; N]::as_mut_ptr()`. What we need instead is `[MaybeUninit<u8>; PATH_MAX as usize]`, and then we can pass `buf.as_mut_ptr().cast::<u8>()`. However, the more I think about this, the more complex it feels, especially when we could declare a `[u8; PATH_MAX as usize]` and initialize it with `[0; PATH_MAX as usize]` instead and avoid any uninitialized values.

Reply:
That's very surprising... Shouldn't they do the same thing and return the pointer to the beginning of the buffer?

Anyway I updated the code to use zero-initialized array to sidestep the issue. This may bring a little overhead but will avoid any potential undefined behaviors...

Reply:
Make sense. I also suspect it's a fat pointer issue...